resources/js/package.js

```diff
@@ -1,37 +1,69 @@
-document.addEventListener('turbo:load', function () {
-    ['shipping_address', 'billing_address'].forEach((type) => {
-        window.app.$watch('checkout.'+type+'.postcode', function () {
-            getAddressFromPostcodeservice(type)
-        })
+import { useThrottleFn } from "@vueuse/core"
+
+const postcodeObserver = new MutationObserver(useThrottleFn(() => {
+    let types = ['housenumber', 'postcode', 'country']
+    types.forEach((type) => {
+        document.querySelectorAll(`[id\$='_${type}']`).forEach(el => {
+            if (el.id in window.app.custom.postcode_data) {
+                return
+            }
 
-        window.app.$watch('checkout.'+type+'.street.1', function () {
-            getAddressFromPostcodeservice(type)
+            window.app.custom.postcode_data[el.id] = el.value
+            let id = el.id.replace(`_${type}`, '')
+            el.addEventListener('change', () => {
+                window.app.custom.postcode_data[el.id] = el.value
+                getAddressFromPostcodeservice(id)
+            })
         })
     })
+}, 500, true, true))
+
+document.addEventListener('turbo:load', function () {
+    window.app.custom.postcode_data = {}
+
+    postcodeObserver.observe(window.app.$el, {
+        childList: true,
+        subtree: true,
+    })
 })
 
-async function getAddressFromPostcodeservice(type) {
-    if (window.app.checkout[type].country_id != 'NL') {
+async function getAddressFromPostcodeservice(id) {
```

Comment:
Is this the right place for all this logic? I feel like in most places it is better to change the data itself instead of the input. Since the input will follow the data

Reply:
We can't do that, unfortunately, which is why I went for this approach to begin with. The checkout and the address edit page are built up completely differently.

As the requirement for this was to have it as a drop-in solution without needing to change any blade files, we do the opposite and we let the data follow the input instead (as everything has a v-model anyway). It's not the cleanest, unfortunately, but it works.